frontend/src/metabase/selectors/data.unit.spec.ts

```diff
@@ -0,0 +1,109 @@
+import { State } from "metabase-types/store";
+import {
+  getHasDataAccess,
+  getHasOwnDatabase,
+  getHasNativeWrite,
+  getHasDatabaseWithJsonEngine,
+} from "./data";
+
+describe("metabase/selectors/data", () => {
+  describe("getHasDataAccess", () => {
+    it("should return true if user has data access", () => {
+      const state = {
+        entities: {
+          databases: { databaseOne: { is_saved_questions: false } },
+        },
+      };
+
+      expect(getHasDataAccess(state as any)).toBe(true);
```

Comment:
You can use `createMockState` to avoid `as any` casting here:

```js
import { createMockState } from "metabase-types/store/mocks";
import { createMockDatabase } from "metabase-types/api/mocks";

it("...", () => {
 const state = createMockState({
   entities: {
      databases: { databaseOne: createMockDatabase() },
   },
 });
});
```

Reply:
Ah awesome, thanks Anton.